runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/SparkGroupAlsoByWindowViaWindowSet.java

```diff
@@ -204,6 +207,34 @@ public void outputWindowedValue(
         this.droppedDueToLateness = droppedDueToLateness;
       }
 
+      /**
+       * Retrieves the timers that are eligible for processing by {@link
+       * org.apache.beam.runners.core.ReduceFnRunner}. Even if a given timer is deemed eligible for
```

Comment:
I'm not sure if the comment on firing is in place here.
I would add the comment you put in `@return` and leave only the first sentence there: `A collection of timers...`

Reply:
👍 
I'll move the part in questions to the call site of `reduceFnRunner.onTimers(...)`.